lesson03/assignment/index.js

```diff
@@ -0,0 +1,144 @@
+    "email": "barbara@example.com",
+    "linkedInUrl": "barbara.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "JavaScript", "C", "Go"
+    ]
+  },
+  {
+    "name": "Javier Hernandez",
+    "jobTitle": "User Experience Engineer",
+    "company": "Web Sites and More",
+    "experience": "5 years",
+    "school": "Seattle University",
+    "major": "Performing Arts",
+    "email": "javier@example.com",
+    "linkedInUrl": "javier.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "CSS"
+    ]
+  },
+  {
+    "name": "Maribel Dominguez",
+    "jobTitle": "Front End Engineer",
+    "company": "Bits and Bytes",
+    "experience": "6 years",
+    "school": "University of Washington",
+    "major": "Mechanical Engineering",
+    "email": "maribel@example.com",
+    "linkedInUrl": "maribel.linkedinprofile.com",
+    "codeLanguages": [
+      "HTML", "CSS", "JavaScript", "React", "Vue", "Redux"
+    ]
+  }
+]
+
+// select DOM node
+const userProfileContainer = document.querySelector(".template-hook");
+
+// Update userProfiles entry counter
+let i =+ 0;
+
+//============forEach DOM NODE FUNCTION============
+//run a forEach on userProfiles array
+userProfiles.forEach(function(el) { //passes each property name/value pair from userProfiles' array of objects to the el paramenter as arguement to function below.
+  //Create new HTML section
+  let nametag = document.createElement('section')
+  nametag.innerHTML = `
+  <h1>${el.name}</h1>
+  `
+
+  //==========Populate nametag section with object properties values from userProfiles.=========
+  //For below, can you automatically ouput property names too?
+  let card__details = document.createElement('ul')
+  card__details.innerHTML = `
+  <li><h3>Job Title: </h3><h4>${el.jobTitle}</h4></li>
+  <li><h3>Company: </h3><h4>${el.company}</h4></li>
+  <li><h3>Experience: </h3><h4>${el.experience}</h4></li>
+  <li><h3>Education: </h3><h4>${el.school}</h4></li>
+  <li><h3>Degree: </h3><h4>${el.major}</h4></li>
+  <li><h3>Email: </h3><h4>${el.email}</h4></li>
+  <li><h3>LinkedIn: </h3><h4>${el.linkedInUrl}</h4></li>
+  <li><h3>Code Proficiencies: </h3></li>
+  `
+
+  //============Code Proficiencies by Entry. DOM output.============
+  //create nested ul element for code languages
+  let codeEntry = document.createElement('ul')
+
+  //reset x counter (codeLanguages array length)
+  let x = 0;
+  for (x; userProfiles[i].codeLanguages.length > x; x++) { //loop runs for array codeLanguages.length number of times
+    let newCodeLang = document.createElement('li')
+    newCodeLang.innerHTML = `
+    <li><h4>${userProfiles[i].codeLanguages[x]} </h4></li>
+    `
+      card__details.appendChild(codeEntry)
+      codeEntry.appendChild(newCodeLang)
+  }
+  //Append profile to previously selected HTML element, userProfileContainer
+  userProfileContainer.appendChild(nametag)
+  nametag.appendChild(card__details)
+
+  // update i counter (next user profile entry)
+  i++;
```

Comment:
This isn't necessary in a javascript loop, as the increment action is defined when we define the loop itself: 
`for (x; userProfiles[i].codeLanguages.length > x; x++)` Here, you've specified that `x` should start as undefined (oops, should have been set to 0), run until it reaches the value of one less than the array length, and increase by 1 in each loop run. Alternatively, this could have been set to: `for (let x = userProfiles[i].codeLanguages.length; x == 0; x--)` This would start `x` at the value of one less than the array length, run until `x` is zero, and decrease `x` by one in each loop run. This would give you the exact same functionality, just in an opposite syntax. This is a contrived example, but hopefully illustrates how flexible the for loop definition is.